apps/src/blockly/eventHandlers.ts

```diff
@@ -23,34 +28,42 @@ import {BLOCK_TYPES} from './constants';
 // We re-disable any orphan call blocks when the definition block is dragged.
 // This bug is tracked by the Blockly team:
 // https://github.com/google/blockly-samples/issues/2035
-export function disableOrphans(blockEvent) {
+export function disableOrphans(event: Abstract) {
   // This check is for when a block goes from disabled to enabled (value false is enabled).
   // We need to run the check on this event due to the Blockly bug described above.
+  if (
+    event.type !== Blockly.Events.BLOCK_CHANGE &&
+    event.type !== Blockly.Events.BLOCK_MOVE &&
+    event.type !== Blockly.Events.BLOCK_CREATE
+  ) {
+    return;
+  }
+  const blockEvent = event as BlockChange | BlockMove | BlockCreate;
   const isEnabledEvent =
     blockEvent.type === Blockly.Events.BLOCK_CHANGE &&
-    blockEvent.element === 'disabled' &&
-    !blockEvent.newValue &&
-    blockEvent.oldValue;
-  if (!blockEvent.workspaceId) {
+    (blockEvent as BlockChange).element === 'disabled' &&
+    !(blockEvent as BlockChange).newValue &&
+    (blockEvent as BlockChange).oldValue;
+
+  if (!blockEvent.blockId || !blockEvent.workspaceId) {
     return;
   }
+
   const eventWorkspace = Blockly.Workspace.getById(blockEvent.workspaceId);
-  const block = eventWorkspace.getBlockById(blockEvent.blockId);
+  const block = eventWorkspace?.getBlockById(blockEvent.blockId);
   if (
     blockEvent.type === Blockly.Events.BLOCK_MOVE ||
     blockEvent.type === Blockly.Events.BLOCK_CREATE ||
     isEnabledEvent
   ) {
-    if (!blockEvent.blockId) {
-      throw new Error('Encountered a blockEvent without a proper blockId');
```

Comment:
I added an early return if there's no block id instead. It seems better to fail silently if we can't find the block id rather than crash the page.